tests/Microsoft.Identity.Test.Unit.net45/CoreTests/WsTrustTests/WsTrustEndpointTests.cs

```diff
@@ -164,6 +165,39 @@ public void TestWsTrustEndpointUsernamePasswordWsTrust13()
             Assert.AreEqual(expectedMessage, requestMessage);
         }
 
+        [TestMethod]
+        [Description("Assert client info is included in userrealm discovery header")]
+        public void UserrealmDiscoveryHeadersTest()
+        {
+            string endpoint = "https://login.microsoftonline.com/common/userrealm/";
+
+            using (var harness = new MockHttpAndServiceBundle())
+            {
+                var requestContext = RequestContext.CreateForTest(harness.ServiceBundle);
+                var userrealmEndpoint = harness.ServiceBundle.WsTrustWebRequestManager.CreateUserrealmEndpoint(endpoint, _username, requestContext);
+
+                Assert.IsNotNull(userrealmEndpoint);
+
+                string[] pathSegments = userrealmEndpoint.Uri.AbsoluteUri.Substring(0).Split(
+                    new[]
+                    {
+                        '='
+                    });
+
+                string[] pathSegments2 = userrealmEndpoint.Uri.AbsoluteUri.Substring(0).Split(
+                    new[]
+                    {
+                        '&'
+                    },
+                    StringSplitOptions.RemoveEmptyEntries);
+                
+                Assert.AreEqual($"{endpoint}{_username}?api-version", pathSegments[0]);
+                StringAssert.Contains(pathSegments2[1], "x-client-SKU");
+                StringAssert.Contains(pathSegments2[2], "x-client-Ver");
+                StringAssert.Contains(pathSegments2[3], "x-client", "Can be either x-client-CPU or x-client-OS depending on where the test runs");
```

Comment:
You shouldn't hardcode the path segments becauyse you cannot guarantee the order. By default the dictionary you created uses a hash table, which is non-deterministic (well .. the test will run fine on your machine, but might be different on other machines). 

Have a look at CollectionAsserts or do a bit of Linq magic to dynamically verify the path segments.